protocol/x/clob/keeper/deleveraging.go

```diff
@@ -152,16 +155,22 @@ func (k Keeper) CanDeleverageSubaccount(
 		satypes.Update{SubaccountId: subaccountId},
 	)
 	if err != nil {
-		return false, err
+		return false, false, err
 	}
 
-	// Deleveraging cannot be performed if the subaccounts net collateral is non-negative.
-	if bigNetCollateral.Sign() >= 0 {
-		return false, nil
+	clobPairId, err := k.GetClobPairIdForPerpetual(ctx, perpetualId)
+	if err != nil {
+		return false, false, err
 	}
+	clobPair := k.mustGetClobPair(ctx, clobPairId)
+
+	// Can deleverage subaccount if net collateral is negative or if the market is in final settlement.
+	canDeleverageSubaccount = bigNetCollateral.Sign() == -1 || clobPair.Status == types.ClobPair_STATUS_FINAL_SETTLEMENT
```

Comment:
nit: these two `bool` should be mutually exclusive. let's move this to line 160 (before `k.GetClobPairIdForPerpetual` to save us from 1 state read if TNC < 0